### DISCUSSION TITLE
Can't read in HATCH entities for a .dxf file

### DISCUSSION THREAD

**Discussion by Lukeor** [DomCR/ACadSharp#302] [Q&A]
Hello,

Is it possible to read in HATCH entities for a DXF file?

I have a very simply DXF file that contains 4 HATCH entities (due to being polygons with a fill colour).

However, when I read in the file using the latest ACadSharp version I instead get them as LWPOLYLINE entities. The Color property of these entities also do not display the correct rgb colour code of the fill colours and instead just have rgb(0,0,0) - black.
The layer names of these entities also do not get read in and instead uses a layer name of '0'.

Apart from that - I've been very pleased with the library. Thank you very much for developing it!

I have attached the DXF file in question to help with any investigations.

Many thanks,
Luke

[DEFAULT_PROJECT0_Export.zip](https://github.com/DomCR/ACadSharp/files/14514833/DEFAULT_PROJECT0_Export.zip)

**Comment by DomCR** [comment]
Hi @Lukeor,

I've been looking at the file, the `DxfReader` is missing the feature the read `Hatch.BoundaryPath.Polyline` and seems that has an error when reading true colors, I'll open an issue to fix it.

About the Layers, looking at the file the LWPOLYLINE that you are using to bound the HATCH are drawn in the Layer 0 so they are assigned correctly, the only thing missing are the HATCH and that's due the missing implementation.

Thanks for reporting this issue to the project!

**Reply by DomCR** [reply]
- https://github.com/DomCR/ACadSharp/issues/303

**Reply by Lukeor** [reply]
No worries at all! Many thanks for taking the time to look into this issue.

With regards to the layer names, when I open the DXF file in another CAD viewer (I use AutoCAD TrueView), the 4 HATCH polygons belong to a layer to 'Housing Polygons' but when read via ACadSharp they are given the layer name '0'. Although I guess the bug you have identified could be the reason why the layer names are also not getting read in correctly?

Here is a picture of the attached DXF file in TrueView (I have turned off the '0' layer so that only the 'Housing' layer is visible):
![image](https://github.com/DomCR/ACadSharp/assets/162500307/a27401bc-6dd2-4538-84cd-cf37321f7668)

Again. thanks for all of your help and I'm loving the project!

Luke

**Reply by DomCR** [reply]
Here is my analysis:

I've selected a Hatch and it's boundary:
![image](https://github.com/DomCR/ACadSharp/assets/85741049/8e8ec5b8-d114-44c5-a0ad-8ff83ea98158)

There is a Polyline with the Hatch, if I isolate the entity I can see that is assigned to the 0 Layer:
![image](https://github.com/DomCR/ACadSharp/assets/85741049/2b7a1771-999d-4016-882a-892cc05d3385)

Also in the DXF you can read the code 8, which stands for the Layer name:

![image](https://github.com/DomCR/ACadSharp/assets/85741049/bfeaf410-3592-4d54-a5e8-8ca0ef713e8f)

[Link](https://help.autodesk.com/view/OARX/2021/ENU/?guid=GUID-3610039E-27D1-4E23-B6D3-7E60B22BB5BD) to the docs.

So for each Hatch there is a Polyline in the Layer 0, that is what you are finding in the `CadDocument` after the reader, the only thing missing are the Hatch, because they are discarded due the missing implementation, once the reader returns the Hatch correcly they will be assigned to the layer "Housing".

Let me know if this helps or I've missed something.

**Reply by Lukeor** [reply]
Ah that makes sense!

Thanks very much for clarifying that for me. I was wondering why there was a '0' layer but your analysis on it being the boundary clears that up.

I've only just recently began development on handling CAD files in my GIS application, so it's very much been a learning curve for me.

Thanks once again for your help.
